plugins/GeoIP/class.geoip.import.php

```diff
@@ -0,0 +1,683 @@
+<?php
+
+class GeoipImport {
+
+    private static $csvDownloadURL = "http://geolite.maxmind.com/download/geoip/database/GeoLite2-City-CSV.zip";
+
+    private static $blockFileName    = 'GeoLite2-City-Blocks-IPv4.csv';
+    private static $locationFileName = 'GeoLite2-City-Locations-%s.csv';
+    private static $locationLocales  = ['de','en','es','fr','ja','pt-BR','ru','zh-CN'];
+
+    const tuncateQuery  = "TRUNCATE `%s`";
+    const describeQuery = "DESCRIBE `%s`";
+    const dropTableQuery = "DROP TABLE %s IF EXISTS";
+
+    private static $blockRowsPerQuery    = 2000;
+    private static $locationRowsPerQuery = 500;
+
+    private $blockImportTime      = 0;
+    private $locationImportTime   = 0;
+
+    private static $blockTableName       = 'geoip_block';
+    private static $locationTableName    = 'geoip_location';
+
+    public $downloadZip;
+    public $locationFile;
+    public $blockFile;
+
+    public  $baseDir = '/tmp/geoip';
+    private $userDir;
+
+    private static $errorLog = "/tmp/geoip.log";
+
+
+    function __construct() {
+
+        // Set user's import directory:
+        $this->userDir = $this->baseDir.'_'.GDN::session()->UserID;
+
+        return;
+    }
+
+
+    /**
+     * Runs primary import procedure for importing GeoIP CSV files into MySQL.
+     *
+     * @return bool Returns TRUE on success, FALSE on failure.
+     */
+    public function run() {
+
+        ini_set('max_execution_time', 600);
+
+        $oldErrorOn  = ini_set("log_errors", true);
+        $oldErrorLog = ini_set("error_log", self::$errorLog);
+
+        error_log(">> ...Starting GeoIP CSV Import... <<");
+        error_log("Log File: ".self::$errorLog, E_USER_NOTICE);
```

Comment:
error_log()'s 2nd argument is not an error level, it is a log destination and has a max value of 4, E_USER_NOTICE = 1024